### DISCUSSION TITLE
Why doesn't SemanticTextMemory.SaveSaveReferenceAsync store the text?

### DISCUSSION THREAD

**Discussion by xbotter** [microsoft/semantic-kernel#438] [Q&A] [answered]
Based on the source code, 
https://github.com/microsoft/semantic-kernel/blob/ccb9c26039f5562be22242354a5639116ff5ada5/dotnet/src/SemanticKernel/Memory/SemanticTextMemory.cs#L49-L58
I found that the method generates an embedding based on the text input, but it doesn't actually store the text in the MemoryRecord. As a result, when I try to search for the text, I can't retrieve it from the MemoryRecord. I'm not sure why it was designed this way and would appreciate some clarification.

**Comment by awharrison-28** [comment, marked as answer]
Hi @xbotter.

**Context:** there are 2 save methods associated with SemanticTextMemory - `SaveReference` and `SaveInformation`.

`SaveReference` is intended to save information from a known source  - this way you can store an embedding and recreate it from the source without having to take up space also storing the source text. 

`SaveInformation` is intended to save information where the source text cannot be referenced. 

**Scenario Examples:**
`SaveReference(
collection: "Semantic Kernel GitHub",
externalId: "https://github.com/microsoft/semantic-kernel/discussions/438", 
sourceName: "Semantic Kernel Discussions", 
text: text,
description: "Why doesn't SemanticTextMemory.SaveReference store the text?")` 

> Where the 'text' might be of a summary of this discussion generated by a semantic skill. 


`SaveReference(
collection: "Semantic Kernel GitHub",
externalId: "https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/SemanticKernel/Memory/SemanticTextMemory.cs", 
sourceName: "SemanticTextMemory.SaveReference", 
text: text,
description: "lines 49-66")` 

> Where the 'text' is the codeblock defining SemanticTextMemory.SaveReference. 

vs


`SaveInformation(
collection: "About Me",
id: "myOccupation", 
text: "Software Engineer")` 

I hope this helps!

**Reply by xbotter** [reply]
Thanks for your explanation, I think I got it now!

**Reply by afederici75** [reply]
Thanks for the answer!
